### PR TITLE
Biomes: Add tundra lowland with permafrost, stones, moss and snow 

### PR DESCRIPTION
**Pull request by paramat**
Remove unsuitable snowblocks from tundra highland, use snow instead.
////////////////

screenshot_20180429_180258

Resurrecting #1261 

Current stony tundra has unsuitable deep snow and is missing permafrost and moss. The new biome is placed at low altitudes and with the same heat/humidity points, the previous tundra is moved to higher land.
Moss and snow are patchy, using 2D noise.

### CODE REVIEW

**Comment by paramat** [comment]
screenshot_20180314_214839

As an alternative, green rolling moss.

**Comment by paramat** [comment]
screenshot_20180314_221215

Or brown rolls of moss.

**Comment by Ezhh** [comment]
The brown (last one you posted) is the best looking to me so far, but I'm not sure if it feels right to have moss everywhere. Feels like it should be more patchy with permafrost and/or stone showing through more.

**Comment by paramat** [comment]
Good suggestion.

**Comment by paramat** [comment]
screenshot_20180315_025819

Does look good, need to fix vertical biome blend now.

**Comment by Fixer-007** [comment]
Textures are controversial for me, especially when mixed with snow.

**Comment by paramat** [comment]
It's meant to be blobby moss and is quite a good texture, however the blobs should probably be green. Maybe i'll make them rarer and add a second surface type, and much more bare stone and permafrost.

**Comment by Ezhh** [comment]
The green texture you tried was really the wrong colour green to my eyes. 

But more generally, it could be very difficult to get this kind of blobbed texture to look right when it needs to tile, because to work it has to look somewhat random and irregular. And like you said, more stone/permafrost has to show through somehow.

**Comment by Fixer-007** [comment]
Maybe take a mossy cobble as a base, and increase amount of green part in it gradually

**Comment by paramat** [comment]
The green can be tuned, what is your suggestion for tuning the colour? These blobs do look better green, they are meant to be moss after all. It looks better being more noisy to break up the continuous moss areas, here moss is rarer too:

screenshot_20180316_003800

Work continuing.

**Comment by paramat** [comment]
The bare permafrost is too bare, might create a stony surface texture as a surface node.

**Comment by cx384** [comment]
I am not sure but shouldn't permafrost belong to group soil and growing of permafrost moss would be nice (it should grow very slow). Also gravel (or cobble) rings and scattered stones would be nice.
https://en.wikipedia.org/wiki/Permafrost#Landforms



<details>
  <summary>more pictures</summary>





</details>

**Comment by paramat** [comment]
Not sure about group soil, as most plants won't grow on it and it's frozen so can't be hoed for farming.
Note that this biome is meant to be winter tundra, not summer when the surface thaws and plants grow on it.

**Comment by paramat** [comment]
Added a dedicated surface node 'permafrost with stones', added a new screenshot to first post. Ready for review.

**Comment by paramat** [comment]
Maybe the stones are a little too bright and could be darker and dirtier.

**Comment by paramat** [comment]
Seems to have a bug ..

File: mods/default/mapgen.lua [hunk 1]
```diff
@@ -948,11 +948,25 @@ function default.register_biomes(upper_limit)
...
+
+	minetest.register_biome({
+		name = "tundra",
+		node_top = "default:permafrost_with_stones",
+		depth_top = 1,
+		node_filler = "default:permafrost",
+		depth_filler = 1,
+		node_riverbed = "default:gravel",
+		depth_riverbed = 2,
+		y_max = 32,
```
[the remark this answers is not in the document]
Comment (paramat) [reply mods/default/mapgen.lua]: Fine, i wasn't sure where to put it, 48?

[the remark this answers is not in the document]
Comment (Ezhh) [reply mods/default/mapgen.lua]: Would be better. :)

**Comment by paramat** [comment]
stones_wrap

^ Wrap texture

stones_wrap_less

^ Wrap with fewer stones

File: mods/default/mapgen.lua [hunk 2]
```diff
@@ -948,11 +948,25 @@ function default.register_biomes(upper_limit)
...
 		depth_riverbed = 2,
 		y_max = upper_limit,
+		y_min = 33,
+		heat_point = 0,
+		humidity_point = 40,
+	})
+
+	minetest.register_biome({
+		name = "tundra",
+		node_top = "default:permafrost_with_stones",
```
[the remark this answers is not in the document]
Comment (paramat) [reply mods/default/mapgen.lua]: Fixed.

**Comment by paramat** [comment]
Updated with 'wrap fewer stones' texture. Moved highland tundra to y = 48. Added texture credits.
Mip mapping might help with that visual noise.

**Comment by Ezhh** [comment]
Fine with this now :+1: 

(mip mapping just disguises it really well beneath excessive and very ugly blurriness, that can also distort colours beyond belief. It's unusable to me, in part because the range it blurs from is even less than the range you can place things at in creative... and it shouldn't be needed for things like this. Either way, this is not an MTG issue.)

**Comment by paramat** [comment]
Yes i forgot how bad mipmap is currently.

**Pull request merged**